### PR TITLE
fix: Fix theme migration logic, follow-up to #819

### PR DESCRIPTION
**Pull request by Rob--W**
See commit message.

This is a follow-up to #819.

### CODE REVIEW

File: src/lib/themes-test.js [hunk 1]
```diff
@@ -33,6 +33,62 @@ describe("lib/themes", () => {
         expect(output).to.deep.equal(theme);
       });
     });
+    it("should migrate themes without popup/popup_text", () => {
+      const theme = {
+        images: {
+          headerURL: ["dummy.png"],
```
Comment (rpl) [inline src/lib/themes-test.js]: shouldn't images.headerURL also be in the expected theme along with "images.theme_frame"?

Comment (Rob--W) [reply src/lib/themes-test.js]: Yes. Apparently I need one of the files in `src/images/patterns` for the test to pick it up. I've updated the test.

**Comment by Rob--W** [comment]
> at a first glance it seems to ne that the "legacy LWT aliases" theme properties are now going to still be part of the theme also in the new version of the extension, am I right?
>
> If that is the case, it may be reasonable to still remove them on the extension side to avoid logging warnings about them in Firefox >= 70 (mostly because they may be misleading).

This is not the case. `convertToBrowserTheme` is used to create the theme for use in the `browser.theme` API. These use the return value of `normalizeTheme`, and are independent of the original input parameter.

I've added a unit test (the first one!) that verifies that `convertToBrowserTheme` does what it's supposed to be doing.

File: src/lib/themes-test.js [hunk 2]
```diff
@@ -33,6 +33,137 @@ describe("lib/themes", () => {
...
+        },
+        title: ["old theme without popup"],
+      };
+      const output = subject(theme);
+      expect(output).to.deep.equal(expectedTheme);
+    });
+    it("should migrate deprecated theme properties", () => {
+      const deprecatedTheme = {
+        images: {
+          headerURL: ["dummy.png"],
```
Comment (rpl) [inline src/lib/themes-test.js]: headerURL is meant to be a string

also, now I see that only background images that are included in bgImages are meant to be still part of the normalized theme, would you mind to add an inline comment on top of this line to mention that it is expected that dummy.png will not be part of `images.additional_backgrounds` in the normalized theme?

File: src/lib/themes-test.js [hunk 3]
```diff
@@ -33,6 +33,146 @@ describe("lib/themes", () => {
         expect(output).to.deep.equal(theme);
       });
     });
+    it("should migrate themes without popup/popup_text", () => {
```
Comment (rpl) [inline src/lib/themes-test.js]: Nit, if you end up to force push / update this PR again, let's add an empty line between the `it(...)` calls (don't bother if you are not going to update it again for now).

Comment (Rob--W) [reply src/lib/themes-test.js]: Added a few blank lines for consistency.

**Comment by caitmuenster** [comment]
Testing prod extension on https://mozilla.github.io/: 

Color-Properties-Fixed-Dev

**Pull request merged**